packages/core/src/controllers/user.ts

```diff
@@ -0,0 +1,342 @@
+import * as rm from 'typed-rest-client';
+import {
+  AddAssignedRoleToUserArguments,
+  AssignedRoleAPIResponse,
+  CreateUserArguments,
+  DecodedToken,
+  GetOrgMembershipsForUserAPIResponse,
+  GetOrgMembershipsForUserArguments,
+  GetUserByIDArguments,
+  GetUsersArguments,
+  GetUsersResponse,
+  HasRoleAPIResponse,
+  HasRoleArguments,
+  HasRoleTokenArguments,
+  HasRoleWalletArguments,
+  PutUserMetadataArguments,
+  RemoveAssignedRoleFromUserArguments,
+  UserResponse,
+  ValidateTokenAPIResponse,
+  ValidateTokenArguments,
+  ValidateTokenResponse,
+} from '@slashauth/types';
+import { signBody, signQuery } from '../utils/query';
+import { base64Decode } from '../utils/strings';
+import { Controller } from './controller';
+import { getBaseURL } from '../utils/url';
+
+// TODO: Need to add:
+// - How do we add a user to an org?
+export class UserController extends Controller {
+  constructor(
+    client_id: string,
+    client_secret: string,
+    apiClient: rm.RestClient
+  ) {
+    super(client_id, client_secret, apiClient);
+  }
+
+  async validateToken({
```

Comment:
We should consider moving this to a shared package because we use it in at least two places.

Reply:
Not as part of this PR though

Reply:
I'm not following. where would you want to place it?

we use it in the two middleware packages and from the direct client. the middleware packages have access to the core package

Reply:
I'm thinking this function could be in some util package that we use across other typescript packages

Reply:
ah okay. got it

the utils package would need access to the client as well but I see what you mean